frontend/src/metabase/common/components/EntityPicker/types.ts

```diff
@@ -24,11 +25,31 @@ export type PickerStateItem<Item, Query> = {
 
 export type EntityPickerOptions = EntityPickerModalOptions;
 
-export type EntityTab<Model extends string> = {
+export type EntityPickerTabRenderProps<
+  Id extends SearchResultId,
+  Model extends string,
+  Item extends TypeWithModel<Id, Model>,
+> = {
+  onItemSelect: (item: Item) => void;
+};
+
+export type EntityPickerTabId = string;
+
+export type EntityPickerTab<
+  Id extends SearchResultId,
+  Model extends string,
+  Item extends TypeWithModel<Id, Model>,
+> = {
+  id: EntityPickerTabId;
```

Comment:
Previously the `model` attribute was being used as a tab id. 
It was confusing, because there is no such model as "recents" or "search".
Now we have a separate attribute for tab id, and separate attribute for the model given tab provides.